### DISCUSSION TITLE
MapVC NavigationController 추가.

### DISCUSSION THREAD

**Discussion by Haroldfromk** [SwiftBreakers/TTEOPPOKKI4U#44] [Ideas]
기존에는 MapVC만 SceneDelegate에서 NavigationController를 활성화 하지 않았는데, 이번에 변경을 했습니다. (아직 push 전)
지도 탭하고 가게정보를 보일때부터 계속 아래에서 올라오는게 좀 그런것 같아서 변경을 해보았습니다. 물론 이게 정답은 아니기에 다시 돌려도 괜찮습니다.
그래서 Ideas에 적어봤어요

구현화면은 다음과 같습니다.

![Jun-15-2024 21-42-32](https://github.com/SwiftBreakers/TTEOPPOKKI4U/assets/97341336/0f9d9088-df7a-45c8-b1df-8d70f8263552)

[Code수정]

코드 수정은 다음과 같습니다.

특히 포인트는 DetailVC입니다.

이전에 말씀드렸던건 NavigationController가 없는 상태였기에 `let mapVC = self?.tabBarController!.selectedViewController as! MapViewController` 가 가능했습니다.
하지만 NavigationController가 씌워지면 달라집니다.

아래 사진을 보시면 MapVC 이전에 NavigationController가 있습니다.

![CleanShot 2024-06-15 at 21 54 07@2x](https://github.com/SwiftBreakers/TTEOPPOKKI4U/assets/97341336/86e4d412-622b-4fec-b3ad-ea4152f75621)

그러므로 Tabbar에서 선택된 VC는 NavigationController가 되고.
거기서 NavigationController의 ViewController중 첫번째를 가져오는데 그걸 MapVC로 다운캐스팅을 해서 인스턴스화 한거로 생각하시면됩니다.

SceneDelegate
```swift
// before
let mapVC = MapViewController()

// after
let mapVC = UINavigationController(rootViewController: MapViewController())
```

DetailViewController
```swift
// before
@objc func moveToMap() {
        
        guard let keyword = card?.title else { return }
        
        NetworkManager.shared.fetchAPI(query: keyword) {[weak self] stores in
            
            if let tabBarController = self?.tabBarController {
                tabBarController.selectedIndex = 1
                
            }
            
            let mapVC = self?.tabBarController!.selectedViewController as! MapViewController
            let location = CLLocation(latitude: Double(stores[0].y)!, longitude: Double(stores[0].x)!)
            mapVC.centerMapOnLocation(location: location)
            
        }
    }


// after
@objc func moveToMap() {
        
        guard let keyword = card?.title else { return }
        
        NetworkManager.shared.fetchAPI(query: keyword) {[weak self] stores in
            
            if let tabBarController = self?.tabBarController {
                tabBarController.selectedIndex = 1
                
                if let navController = tabBarController.selectedViewController as? UINavigationController,
                   let mapVC = navController.viewControllers.first as? MapViewController {
                    
                    let location = CLLocation(latitude: Double(stores[0].y)!, longitude: Double(stores[0].x)!)
                    mapVC.centerMapOnLocation(location: location)
                }
            }
            
        }
    }
```

**Comment by jinmoon23** [comment]
[SceneDelegate]
`let mapVC = UINavigationController(rootViewController: MapViewController()) `
[DetailVC]
`@objc func moveToMap() {
        
        guard let keyword = card?.queryName else { return }
        
        NetworkManager.shared.fetchAPI(query: keyword) {[weak self] stores in
            
            guard let tabBarController = self?.tabBarController else { return }
                   tabBarController.selectedIndex = 1
            
            if let navController = tabBarController.selectedViewController as? UINavigationController,
               let mapVC = navController.viewControllers.first as? MapViewController {
                mapVC.searchLocation(query: self!.card!.queryName)
                mapVC.storeInfoView.isHidden = false
            }
            
        }
    }`

수정 후 정상작동 확인하였습니다!
